### PR TITLE
[CSM-402] Avoid repeated inputs in transactions from history

### PR DESCRIPTION
**Pull request by Martoon-00**
I actually had no idea how would I test it, since this is reproduced very rarely.

But somehow I got following transaction formed:
```
[wallet-api:INFO:ThreadId 2670] [2017-08-04 22:12:13 MSK] Submitting transaction: Tx 0b0dd9c0 with inputs [TxIn e6b95974 #1, TxIn e68c0c5d #1, TxIn e56d9549 #1, TxIn e1ce44f8 #1, TxIn c9fc6a84 #1, TxIn c66
56f4d #1, TxIn bfb1a7f7 #1, TxIn b693d777 #1, TxIn 81ff481d #1, TxIn 69b77dbf #1, TxIn 68d5785f #1, TxIn 62d457ec #1, TxIn 5348a434 #1, TxIn 3ff22ae2 #1, TxIn 39ded5f0 #1, TxIn 2ff27c56 #1, TxIn 2ce089c1 #
1, TxIn 1a25f38b #1, TxIn 070c9a8c #1, TxIn 027f0f6d #1], outputs: [TxOut 17000 coin(s) -> 19HN6YWz5K3uqaf27Z161NgPXDmgy4igiiPajovkox6PdqWfDriFiyhHPiYfxaormZKbRGxXtRroyUFmsxQwuBQnNqsKuG, TxOut 3000 coin(s)
 -> 19HA3RibW3jCqX5QgBkTtnyei76sCCDqAvbBqU19xoYyS1h7soJGLD1k5o61mrNQY82dBYDpPuax5AoaDWbPSD6VKuA86y]
witnesses: [
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>,
    PkWitness: key = pub:99b70e6f, key hash = 5ac700f3, sig = <signature>
]
distribution: [[], []]
```
which displayed correctly in history, so I think that's a proof

### CODE REVIEW

**Comment by Martoon-00** [comment]
I failed to pass daedalus acceptance tests on obviously dummy stages, can try again when this process gets arranged

File: src/Pos/Wallet/Web/Tracking.hs [hunk 1]
```diff
@@ -332,15 +332,15 @@ trackingApplyTxs (getEncInfo -> encInfo) allAddresses getDiff getTs txs =
             tx@(UnsafeTx (NE.toList -> inps) (NE.toList -> outs) _) = taTx
             txId = hash tx
 
-        resolvedInputs <- catMaybes <$> mapM (\tin -> fmap (tin, ) <$> utxoGet tin) inps
+        resolvedInputs <- catMaybes <$> mapM (\tin -> (tin, ) <<$>> utxoGet tin) inps
         let txOutgoings = map txOutAddress outs
-            txInputs = map (toaOut . snd) resolvedInputs
+            txInputs = mergeTxOuts $ map (toaOut . snd) resolvedInputs
```
Comment (flyingleafe) [inline src/Pos/Wallet/Web/Tracking.hs]: You could simply `ordNub` the `inps` before, and you'll guarantee that no resolved inputs or outputs are the same.

**Comment by flyingleafe** [comment]
I still don't understand how such transactions can occur - aren't they banned by verification?
I strongly hope that one cannot actually _double spend_ using such transactions...
Anyway, I have minor consideration, check it out.

File: src/Pos/Client/Txp/History.hs [hunk 2]
```diff
@@ -118,9 +130,9 @@ getTxsByPredicate pr mDiff mTs txs = go txs []
   where
     go [] acc = return acc
     go ((wh@(WithHash tx txId), _wit, dist) : rest) acc = do
-        inputs <- getSenders tx
+        inputs <- mergeTxOuts <$> getSenders tx
```
Comment (flyingleafe) [inline src/Pos/Client/Txp/History.hs]: Waaait a minute, so if here transaction spends several outputs which belong to _one_ address, here we'll get that transaction has only one input resolved? But this simply _isn't true_ -- we should display _real_ outputs.

**Comment by Martoon-00** [comment]
Close for the lack of sense

**Pull request closed without merging**